README.md

```diff
@@ -8,152 +8,62 @@ described [here](https://github.com/wangkuiyi/k8s-ml), which makes use
 of Kubernetes to run jobs in Docker containers.  An ideal platform to
 run Kubernetes is CoreOS.  However, CoreOS doesn't include CUDA GPU
 driver in its kernel.  In order to use GPUs for accelerated deep
-learning, we build CUDA GPU driver as kernel modules, and load them in
-Docker images of Tensorflow, Torch and other programs that relies on
-CUDA GPU drivers.  At runtime, we load CUDA driver kernel modules from
-within Docker containers.
+learning, we want to build CUDA GPU driver as kernel modules and to
+load them in Docker images together with Tensorflow, Torch and other
+programs that relies on CUDA GPU drivers.  At runtime, we load CUDA
+driver kernel modules from within Docker containers.
 
 ## Solution
 
-Thanks https://github.com/emergingstack/es-dev-stack, which shows that
+https://github.com/emergingstack/es-dev-stack shows that
 above idea works.  The problem with the `es-dev-stack` tool is that
 the provided Dockerfile builds too big CUDA Docker images.  Actually,
 the disk space of an AWS EC2 g2.8xlarge instance doesn't support build
-and run such images.  Our solution is a two-phase approach:
+and run such images.
 
-1. The `builder` image downloads necessary Linux kernel and CUDA Toolkit
-   and build CUDA drivers as kernel modules.  `builder` can run on a
-   VirtualBox VM with enough big disk spaces.  The host of the VM
-   doesn't need to have CUDA GPU.  `builder` is derived from
-   `ubuntu:14.04` image.  `builder` writes two kernel modules files:
+So I use a two-phase approach: build a Docker image which build CUDA
+kernel drivers at runtime, and build application images that loads
+CUDA drivers and GPU applications.  This repo focuses on the first
+phase.
 
- 1. `nvidia.ko`
- 1. `nvidia-uvm.ko`
+## Build
 
-1. Some `cuda` images, each based on Tensorflow image or Torch image,
-   as well as the two kernel modules.  The Dockerfile runs command
-   like `insmod /opt/nvidia.ko && insmod /opt/nvidia-uvm.so` to load
-   kernel modules before starting programs that rely on GPU.  These
-   `cuda` images are supposed to run on computers with Docker and GPU.
+The primary challenge is that the building process needs CUDA Toolkit
+and Linux kernel source code, both take huge amount of disk space.  My
+solution is to use the disk space of the host computer (my iMac 5K) as
+much as possible.
 
+My iMac doesn't run CoreOS, so I need to run a virtual machine with
```

Comment:
Ah I see! This is in the build process! Alternatively you can do the build from docker run instead of docker build and get the artifact from mounted host-dir . Docker run -v host-dir:/build-output coreos-image build-driver.sh --rm build.sh